core/frontend/src/components/health/HealthTrayMenu.vue

```diff
@@ -205,7 +205,7 @@ export default Vue.extend({
   },
   mounted() {
     mavlink2rest.startListening('HEARTBEAT').setCallback((message) => {
-      if (message?.header.system_id !== 1 || message?.header.component_id !== 1) {
+      if (message?.message.mavtype.type === 'MAV_TYPE_GCS' || message?.header.component_id !== 1) {
```

Comment:
I'm not certain what this `mounted` function is for. Is it supposed to indicate that the vehicle is connected to a topside computer, or is it supposed to show that an autopilot is connected and running well, or something else?

Does the current non-zero component ID check mean that being connected to the onboard camera manager counts as being mounted?

Reply:
this makes the Heartbeat icon beat. This change makes it beat with anything sending a heartbeat that is not a GCS.
We should filter for all valid types here, too.

Reply:
Ahh, I think I misunderstood the logic - your description clarified what it's actually doing, thanks :-)

That said, I don't think the heartbeat icon should beat if only the camera manager is present, so it should only be beating if there's a valid autopilot connected. Presumably that's what you meant by filtering for valid types?

Reply:
done